core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -647,32 +647,37 @@ struct ViewOffset<
            m_dim.N5 * m_dim.N6;
   }
 
-  // Stride with [ rank ] value is the total length
   template <typename iType>
-  KOKKOS_INLINE_FUNCTION void stride(iType* const s) const {
+  KOKKOS_INLINE_FUNCTION void stride_fill(iType* const s) const {
     s[0] = 1;
-    if (0 < dimension_type::rank) {
+    if constexpr (0 < dimension_type::rank) {
       s[1] = m_dim.N0;
     }
-    if (1 < dimension_type::rank) {
+    if constexpr (1 < dimension_type::rank) {
       s[2] = s[1] * m_dim.N1;
     }
-    if (2 < dimension_type::rank) {
+    if constexpr (2 < dimension_type::rank) {
       s[3] = s[2] * m_dim.N2;
     }
-    if (3 < dimension_type::rank) {
+    if constexpr (3 < dimension_type::rank) {
       s[4] = s[3] * m_dim.N3;
     }
-    if (4 < dimension_type::rank) {
+    if constexpr (4 < dimension_type::rank) {
       s[5] = s[4] * m_dim.N4;
     }
-    if (5 < dimension_type::rank) {
+    if constexpr (5 < dimension_type::rank) {
       s[6] = s[5] * m_dim.N5;
     }
-    if (6 < dimension_type::rank) {
+    if constexpr (6 < dimension_type::rank) {
       s[7] = s[6] * m_dim.N6;
     }
-    if (7 < dimension_type::rank) {
+  }
+
+  // Stride with [ rank ] value is the total length
+  template <typename iType>
+  KOKKOS_INLINE_FUNCTION void stride(iType* const s) const {
+    stride_fill(s);
+    if constexpr (7 < dimension_type::rank) {
       s[8] = s[7] * m_dim.N7;
```

Comment:
stride fill is a new function. It only fills up to rank entries. stride(..) fills one more for the total length ...

Reply:
Basically what Christian said, the original `stride` function gives the ending stride one past the length. It caused an array overrun bug in the previous implementation, so I had to add this for better compatibility with mdspan. I tried to re-use as much of the old code as possible and not break it because we depend on it elsewhere.